Chapter 1/2_Check Permutation/CheckPermutation.py

```diff
@@ -7,14 +7,8 @@ def check_permutation(string):
     str2 = string[1]
     if len(str1) != len(str2):
         return False
-    counter = Counter()
```

Comment:
Please see - https://github.com/careercup/CtCI-6th-Edition-Python/pull/129

I think that what you provided is a great usage of python built-in, but elsewhere could use some tuning up, and ultimately maybe goes against the spirit of the interview. If you apply the 2 Counters equality check, you can probably eliminate lines 8 and 9, checking for length equality, since the Counter equality should do this inherently. 

It is also to be noted that you are forced to iterate through the entire array, whilst the solution stops the moment an inequality is encountered.

Thus, I like https://github.com/careercup/CtCI-6th-Edition-Python/pull/129 because it demonstrates the use of Counter() that an interviewer may be impressed by, optimizes (slightly) the population of the Counter, but also maintains the logic in the original solution.

Arguable of course!